src/contexts/walletconnect.tsx

```diff
@@ -144,7 +134,8 @@ export const WalletConnectContextProvider: FC = ({ children }) => {
             gasAmount: p.gasAmount,
             gasPrice: p.gasPrice
           }
-          setTxData(['transfer', txData])
+          setDappTransactionData(txData)
+          openSendTxModal('transfer')
```

Comment:
In my previous approach, it was impossible to set the TxModal as 'deploy-contract' by accident due to types.

Reply:
I don't get your point. wallet connect should be able to open a `deploy-contract` type of send tx modal. (line 142)

Reply:
My point was about reducing accidental coding errors.

If you write `deploy-contract` here by accident, the compiler will not complain. If you type `setTxData(['deploy-contract', txData])` in my previous code, the compiler will complain.

Reply:
I am fine with your change though.

Reply:
I see, I addressed the issue with a new commit